dash/include/dash/halo/Halo.h

```diff
@@ -0,0 +1,1062 @@
+    // so we rather avoid this comparison:
+    if (this == &other) {
+      return true;
+    }
+#endif
+    if (&_region_view == &(other._region_view) || _region_view == other._region_view) {
+      return gidx_cmp(_idx, other._idx);
+    }
+    // TODO not the best solution
+    return false;
+  }
+
+  std::array<index_type, NumDimensions> glob_coords(index_type idx) const {
+    return _pattern.memory_layout().coords(idx, _region_view);
+  }
+
+private:
+  /// Global memory used to dereference iterated values.
+  GlobMemT& _globmem;
+  /// Pattern that created the encapsulated block.
+  const PatternT& _pattern;
+
+  const ViewSpecT _region_view;
+  /// Iterator's position relative to the block border's iteration space.
+  index_type _idx{0};
+  /// Number of elements in the block border's iteration space.
+  index_type _size{0};
+  /// Maximum iterator position in the block border's iteration space.
+  index_type _max_idx{0};
+  /// Unit id of the active unit
+  dart_unit_t _myid;
+
+  ElementT* _lbegin;
+
+}; // class HaloBlockIter
+
+template <typename ElementT, typename PatternT, typename PointerT, typename ReferenceT>
+std::ostream& operator<<(std::ostream& os,
+                         const RegionIter<ElementT, PatternT, PointerT, ReferenceT>& i) {
+  std::ostringstream ss;
+  dash::GlobPtr<ElementT, PatternT> ptr(i);
+  ss << "dash::HaloBlockIter<" << typeid(ElementT).name() << ">("
+     << "idx:" << i._idx << ", "
+     << "gptr:" << ptr << ")";
+  return operator<<(os, ss.str());
+}
+
+template <typename ElementT, typename PatternT, typename PointerT, typename ReferenceT>
+auto distance(
+    /// Global iterator to the initial position in the global sequence
+    const RegionIter<ElementT, PatternT, PointerT, ReferenceT>& first,
+    /// Global iterator to the final position in the global sequence
+    const RegionIter<ElementT, PatternT, PointerT, ReferenceT>& last) ->
+    typename PatternT::index_type {
+  return last - first;
+}
+
+template <typename ElementT, typename PatternT, dim_t NumDimensions>
+class Region {
+
+private:
+  using GlobMemT        = GlobStaticMem<ElementT, dash::allocator::SymmetricAllocator<ElementT>>;
+  using HaloRegionSpecT = HaloRegionSpec<NumDimensions>;
+  using ViewSpecT       = typename PatternT::viewspec_type;
+  using region_index_t  = typename HaloRegionSpecT::region_index_t;
+  using size_type       = typename PatternT::size_type;
+  using BorderT         = std::array<bool,NumDimensions>;
+
+public:
+  using iterator        = RegionIter<ElementT, PatternT>;
+  using const_iterator  = const iterator;
+
+public:
+  Region(const HaloRegionSpecT& region_spec, const ViewSpecT& region, GlobMemT& globmem,
+         const PatternT& pattern, const BorderT& border)
+      : _region_spec(region_spec), _region(region), _border(border),
+        _border_region(std::any_of(border.begin(), border.end(), [](bool border_dim){ return border_dim == true;})),
```

Comment:
line length >80 chars?

Reply:
120

Reply:
Please stick to 80 :)

Reply:
but we don't have IBM punched cards anymore and instead this nice wide screens :)